src/app/store/integration-support.service.ts

```diff
@@ -41,8 +47,11 @@ export class IntegrationSupportService {
     return this.http.post(url, configuredProperties);
   }
 
-  requestJavaInspection(connectorId: String, type: String) {
-    const url = this.mapperBaseUrl + '/java-inspections/' + connectorId + '/' + type + '.json';
-    return this.http.get(url);
+  requestJavaInspection(connectorId: String, type: String): Observable<Response> {
+    if (this.mapperBaseUrl) {
+      const url = this.mapperBaseUrl + '/java-inspections/' + connectorId + '/' + type + '.json';
```

Comment:
So with this much URL assembling it'd be probably better to use a helper to build the URL, for example:

https://github.com/syndesisio/syndesis-ui/blob/master/src/app/store/integration-support.service.ts#L40

I kinda wonder if we shouldn't consider bringing in https://medialize.github.io/URI.js/

Reply:
Thanks for the comment! can you teach me a bit around it, so the Restangular passed in as a constructor arg seems to be pointing
```https://$(syndesis-rest)/api/v1```
But the java inspection URL is
```https://$(syndesis-rest)/mapper/v1```
is it something I can modify on this existing Restangular instance, or need to instantiate another one here? don't know how to neither...

Reply:
In fact yeah, I'd just suggest using restangular to build this URL instead, there's examples in the same file for other API calls.

Reply:
@igarashitm yeah, I'm kinda thinking maybe `yarn add urijs` and then you can do:

```
   // maybe make mapperUrl a member variable
   const mapperUri = new URI(config.getSettings().apiEndpoint).path('/mapper/v1');
```

and then later on:

```
   const targetUrl = mapperUri.segment('java-inspections').segment(connectorId).segment(type + '.json');
```